packages/main/src/plugin/util/exec.ts

```diff
@@ -0,0 +1,58 @@
+/**********************************************************************
+ * Copyright (C) 2023 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import * as cp from 'child_process';
+import type { RunOptions, RunResult } from '@podman-desktop/api';
+
+export function runCliCommand(command: string, args: string[], options?: RunOptions): Promise<RunResult> {
+  return new Promise<RunResult>((resolve, reject) => {
+    const cmd = `${command} ${args.join(' ')}`;
+
+    const child = cp.exec(
+      cmd,
+      { env: options?.env },
+      (error: cp.ExecException | null, stdout: string, stderr: string) => {
+        const result: RunResult = {
+          exitCode: error ? error.code || 1 : 0,
+          stdOut: stdout.trim(),
+          stdErr: stderr.trim(),
+        };
+
+        resolve(result);
+      },
+    );
+
+    child.on('error', error => {
+      reject(error);
+    });
+  });
+}
```

Comment:
It should take the implementation of podman extension (here it won't work with spaces, it's using exec and not spawn, it won't work in flatpak, etc)

https://github.com/containers/podman-desktop/blob/main/extensions/podman/src/util.ts#L70-L109

and

https://github.com/containers/podman-desktop/blob/main/extensions/podman/src/podman-cli.ts#L64-L125

Reply:
updated, now spawn method is used instead exec under the hood